src/drivers/mct/shr/seq_timemgr_mod.F90

```diff
@@ -1538,13 +1546,21 @@ subroutine seq_timemgr_alarmInit( EClock, EAlarm, option, opt_n, opt_ymd, opt_to
        call ESMF_TimeSet( NextAlarm, yy=cyy, mm=cmm, dd=opt_n, s=0, calendar=seq_timemgr_cal, rc=rc )
 
     case (seq_timemgr_optNSteps)
-       call ESMF_ClockGet(EClock, TimeStep=AlarmInterval, rc=rc)
+       if (present(CplTimeStep)) then
+          AlarmInterval = CplTimeStep
+       else
+          call ESMF_ClockGet(EClock, TimeStep=AlarmInterval, rc=rc)
```

Comment:
Should this abort (with a message like, "for optNSteps, must provide coupler time step") rather than falling back to the incorrect (as I understand it) use of the component's time step?